pkg/kubeconfig/kubeconfig.go

```diff
@@ -12,6 +14,12 @@ import (
 	"github.com/aws/eks-anywhere/pkg/validations"
 )
 
+// Writer reads the kubeconfig secret on a cluster and copies the contents to a writer.
+type Writer interface {
+	WriteKubeconfig(ctx context.Context, clusterName, kubeconfig string, w io.Writer) error
+	GetClusterKubeconfig(ctx context.Context, clusterName, kubeconfigPath string) ([]byte, error)
+}
```

Comment:
Where is this interface required?
If I'm not mistaken the `WriteKubecongig` is required by a few consumers and the `GetClusterKubeconfig` by other

But there is no consumer that needs both. That's why it's better to define the interface where they are consumed when you can (which is most of the time)